### PR TITLE
GetExtendedKey with custom account

### LINKED ISSUE #2084: Add way to derive extended pubkey for custom account (HDWallet)
Method `TWHDWalletGetExtendedPublicKey` derives an extended public key, but always using the default 0 account value (it uses derivation path `m/purpose/coin/0'`.
Add a way to obtain extended public key for non-default accounts.

Triggered by discussion https://github.com/trustwallet/wallet-core/discussions/2081

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Add a way to obtain extended public key for non-default accounts.
Method TWHDWalletGetExtendedPublicKey derives an extended public key, but always using the default 0 account value.

Fixes #2084 .

## Testing instructions

Unit tests.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: src/HDWallet.h [hunk 1]
```diff
@@ -80,11 +80,11 @@ class HDWallet {
     /// Derives the address for a coin.
     std::string deriveAddress(TWCoinType coin) const;
 
-    /// Returns the extended private key.
-    std::string getExtendedPrivateKey(TWPurpose purpose, TWCoinType coin, TWHDVersion version) const;
+    /// Returns the extended private key, derivation path used is "m/purpose'/coin'/account'".
+    std::string getExtendedPrivateKey(TWPurpose purpose, TWCoinType coin, TWHDVersion version, uint32_t account = 0) const;
```
Comment (hewigovens) [inline src/HDWallet.h:84]:
```suggestion
    std::string getExtendedPrivateKey(TWPurpose purpose, TWCoinType coin, TWHDVersion version, uint32_t account) const;
```

File: src/HDWallet.h [hunk 2]
```diff
@@ -80,11 +80,11 @@ class HDWallet {
...
 
-    /// Returns the extended public key.
-    std::string getExtendedPublicKey(TWPurpose purpose, TWCoinType coin, TWHDVersion version) const;
+    /// Returns the extended public key, derivation path used is "m/purpose'/coin'/account'".
+    std::string getExtendedPublicKey(TWPurpose purpose, TWCoinType coin, TWHDVersion version, uint32_t account = 0) const;
```
Comment (hewigovens) [inline src/HDWallet.h:87]:
```suggestion
    std::string getExtendedPublicKey(TWPurpose purpose, TWCoinType coin, TWHDVersion version, uint32_t account) const;
```

**Review by hewigovens — COMMENTED** [review]
Let's not use default value in cpp implementations, I found it's easy to mess up when you add more parameter type later (from the account public key pr)

File: src/HDWallet.h (hunk 1 above)
Comment (catenocrypt) [reply src/HDWallet.h]: With default value method is called in 70+ cases; I've duplicated the CPP method as well, to get rid of default parameter type.

**Review by hewigovens — APPROVED** [review]

**Comment by mark2021blue** [comment]
great news!
@catenocrypt @hewigovens  Thank you very much! ! !

**Pull request merged**